### DISCUSSION TITLE
Die erstellte Rechnungen können nicht validiert werden

### DISCUSSION THREAD

**Discussion by ddobric** [stephanstapel/ZUGFeRD-csharp#483] [General]
Ich habe versucht die Rechnungen mit https://www.portinvoice.com/ zu validieren. Wenn ich versuche die Rechnung auch ../ZUGFeRD-csharp\demodata\zugferd22 zu validieren, ehalte ich folgende Fehlermeldung:

![image](https://github.com/user-attachments/assets/86416c45-45c0-4703-b715-b661529de3b0)

Das gleich passiert, wenn ich versuche eine mit der API erstellte Rechnung zu validieren.

**Comment by stephanstapel** [comment]
kannst Du bitte die xml-Datei teilen?

**Comment by ddobric** [comment]
[zugferd_2p2_EXTENDED_Fremdwaehrung-factur-x.zip](https://github.com/user-attachments/files/18049066/zugferd_2p2_EXTENDED_Fremdwaehrung-factur-x.zip)

![image](https://github.com/user-attachments/assets/8e9a233d-9684-4d1a-9b8e-b69c12c7ea3a)

**Comment by stephanstapel** [comment]
Die Regel findest Du in der xslt-Datei:

![grafik](https://github.com/user-attachments/assets/86a140af-5a86-4fab-9ccb-bae520c88dae)

ist auf jeden Fall interessant, dass eine offizielle ZUGFeRD 2.2-Demodatei sich aktuell nicht mehr validieren lässt. Zeigt die Dynamik des Standards. Vor allem zeigt e sauch, das halte ich für dramatisch, wie verschiedene Validierungsregeln zu totaler Verwirrung führen.

Das verwendete 9958 (German Leitweg ID):

![grafik](https://github.com/user-attachments/assets/2c721c6b-cf46-4ff7-822d-14a0ace70150)

wurde in der gleichen Demo-Datei für ZUGFeRD 2.3 ersetzt durch 0204 (Leitweg ID):

![grafik](https://github.com/user-attachments/assets/6f93c4a7-4d18-43f0-a938-b3db0c931a38)

die anderen Validierungsverstöße habe ich mir nicht angeschaut, ich denke, das verhält sich ähnlich.

**Reply by ddobric** [reply]
Genau wegen der Dynamik habe ich Issue eöffnet. Wie gehen wir hier weiter vor?
Btw. Ich habe die ZugPferd  SDK verwendet, um einige Rechnungen zu erstellen. Auch diese lassen sich nicht validieren.

**Reply by stephanstapel** [reply]
Es bleibt nichts anderes übrig, als ab und zu die Dateien mit aktuellen Regeln zu validieren.

Am Ende wird es so sein, dass man mit den Rechnungsempfängern eine Art Kontrakt vereinbart. Es ist ja nicht nur die Frage des Erstellers, ob (wie in diesem Beispiel) die 0204 statt der 9958 verwendet wird. Auch der Empfänger muss darauf vorbereitet sein und nicht die Leitweg Id unter der SchemeID 9958 auslesen.

Grundsätzlich helfen die Validatoren ganz gut, sich schrittweise voranzutasten zu validen ZUGFeRD- oder XRechnung-Dateien.

**Comment by ddobric** [comment]
Vielen Dank @stephanstapel.
Gibt es eine Möglichkeit in der API bei der Erstellung von Rechnung das Schema zu setzen?

**Reply by stephanstapel** [reply]
Was genau meinst Du?
